pkg/util/webhooks/validating-webhooks/validating-webhook.go

```diff
@@ -71,12 +65,26 @@ func Serve(resp http.ResponseWriter, req *http.Request, admitter admitter) {
 			Kind:       "AdmissionReview",
 		},
 	}
-	ctx := context.Background() // todo: use timeout context with the "timeout" request query parameter
+
+	ctx, cancel, err := getContextFromRequest(req)
+	if err != nil {
+		resp.WriteHeader(http.StatusBadRequest)
+		resp.Write([]byte(err.Error()))
+		return
+	}
+
+	defer func() {
+		if cancel != nil {
+			cancel()
+		}
```

Comment:
What is the scenario on which the `cancel` function could be nil?

Reply:
added comment